internal/painter/gl/shaders/round_rectangle.frag

```diff
@@ -21,29 +19,30 @@ float calc_distance(vec2 p, vec2 b, float r)
 void main() {
 
     vec4 frag_rect_coords = vec4(rect_coords[0], rect_coords[1], frame_size.y - rect_coords[3], frame_size.y - rect_coords[2]);
-    // vec2 rect_size_half =  vec2(frag_rect_coords[1] - frag_rect_coords[0], frag_rect_coords[3] - frag_rect_coords[2]) * 0.5 - vec2(stroke_width_half);
     vec2 vec_centered_pos = (gl_FragCoord.xy - vec2(frag_rect_coords[0] + frag_rect_coords[1], frag_rect_coords[2] + frag_rect_coords[3]) * 0.5);
 
     float distance = calc_distance(vec_centered_pos, vec2(rect_size_half.x, rect_size_half.y), radius - stroke_width_half);
 
     vec4 from_color = stroke_color; //Always the border color. If no border, this still should be set
-    vec4 to_color = vec4(1.0, 1.0, 1.0, 0.0); //Outside color
+    vec4 to_color = stroke_color; //Outside color
+    to_color[3] = 0.0; // blend the stroke colour to alpha
 
-    if (stroke_width_half > 0.0)
+    if (stroke_width_half == 0.0)
     {
-        if (distance < 0.0)
-        {
-            to_color = fill_color;   
-        } 
-        
-        distance = abs(distance) - stroke_width_half;
+        from_color = fill_color;
+        to_color = fill_color;
+        to_color[3] = 0.0; // blend the fill colour to alpha
```

Comment:
It seems that in all case we need to set to_color[3] to zero, in which case it would be good to move it outside and after the if.

Reply:
Not in all cases, no - only 2 out of 3. When the fill is being drawn we want the original colour alpha

Reply:
The code for the shader look like this after this patch:
```
    to_color[3] = 0.0; // blend the stroke colour to alpha

    if (stroke_width_half == 0.0)
    {
        from_color = fill_color;
        to_color = fill_color;
        to_color[3] = 0.0; // blend the fill colour to alpha
    }
```

If you move to_color[3] = 0.0, I think it does exactly the same things as now.
```
    if (stroke_width_half == 0.0)
    {
        from_color = fill_color;
        to_color = fill_color;
    }
    to_color[3] = 0.0; // blend the stroke or fill colour to alpha

```

Reply:
Ah those to, before the third case. I see yes